packages/aws-rfdk/docs/upgrade/upgrading-0.38.md

```diff
@@ -0,0 +1,53 @@
+# Upgrading to RFDK v0.38.x or Newer
+
+Starting in RFDK v0.38.0, [Deadline Secrets Management](https://docs.thinkboxsoftware.com/products/deadline/10.1/1_User%20Manual/manual/secrets-management/deadline-secrets-management.html) will be enabled on the
+`Repository` and `RenderQueue` constructs by default when using Deadline 10.1.18 and later. This can cause some issues with existing farm configurations because Deadline Secrets Management requires:
+
+- The internal protocol on the Render Queue to be HTTPS
+- The external protocol on the Render Queue to be HTTPS
+
+Additionally, if you plan on upgrading to RFDK 0.38 or newer and Deadline 10.1.18 or newer, but do **not** want to enable Deadline Secrets Management, you may have to make changes to your RFDK application so that
+RFDK does not automatically configure Deadline Secrets Management on your farm.
+
+## Upgrading Farms and Enabling Deadline Secrets Management
+
+---
+
+_This section describes how to upgrade your RFDK farm while also enabling Deadline Secrets Management. If you would **not** like to enable Deadline Secrets Management, please skip to the
+next section "Upgrading Farms Without Enabling Deadline Secrets Management"._
+
+---
+
+To determine whether your `RenderQueue` is using HTTPS for internal and external protocols, you need to check what values you have set
+for the [`trafficEncryption`](https://docs.aws.amazon.com/rfdk/api/latest/docs/aws-rfdk.deadline.RenderQueue.html#trafficencryption) property.
+Your `RenderQueue` is using HTTPS for internal and external protocols if one of the following is true:
+
+1. You have not specified the `trafficEncryption` property for the `RenderQueue`
+2. You have specified the `trafficEncryption` property for the `RenderQueue` and the following are true:
+    1. The `internalProtocol` is `ApplicationProtocol.HTTPS`
+    2. One of the following is true for the `externalTLS` property:
+        1. `enabled` is `true`
+        2. There are values for both `acmCertificate` and `acmCertificateChain`
+        3. There is a value for `rfdkCertificate`
```

Comment:
Right, thanks for catching that, and having code samples sounds like a good idea, added. I used some formatting to make them expandable items since the code blocks are quite long, what do you think of it?